src/examples/zkapps/zkapp-self-update.ts

```diff
@@ -9,18 +9,14 @@ import {
   Mina,
   AccountUpdate,
   Provable,
-  TransactionVersion,
 } from 'o1js';
 
 class SelfUpdater extends SmartContract {
   init() {
     super.init();
     this.account.permissions.set({
       ...Permissions.default(),
-      setVerificationKey: {
-        auth: Permissions.proof(),
-        txnVersion: TransactionVersion.current(),
-      },
+      setVerificationKey: Permissions.VerificationKey.proofUntilHardfork(),
```

Comment:
I appreciate the creativity to try get it in as a non-breaking change. It looks kind of awkward though to set that particularly permission differently to the rest.

WDYT about deprecating Impossible (so it still works, and the deprecation points to the new one. E.g.)

/**
 * @deprecated `Impossible` has been deprecated and will be removed in a future version. You should use ImpossibleUntilHardFork
 */
Permissons.Impossible

Reply:
And add Permissions.ImpossibleUntilHardFork

Reply:
It was already looking differently than the rest though!

For any other permission, it's
```ts
Permission.impossible()
```
For verification keys, it used to be
```ts
{
  auth: Permission.impossible(),
  txnVersion: Transaction.current()
}
```

so I think it becomes less awkward with this change

Reply:
Per slack, we resolved on `ImpossibleDuringCurrentVersion` as the most accurate and most descriptive (following the existing casing standards of course)